azerothcore/azerothcore-wotlk
# feat(Core): Allow localized motd

## Changes Proposed:

This PR proposes changes to:

## Issues Addressed:

- Closes #17911 

## SOURCE:

The changes have been validated through:

## Tests Performed:

This PR has been:


## How to Test the Changes:



1. Apply the sql that adjusts the motd table
2. Prefill table with new localized motds and check motd by either logging in / reloading the table / running the ingame command .server motd

## Known Issues and TODO List:


- [x] Default server command is currently only updating enUS locale
- [x] Questions remain about how to handle missing locale data. Fallback to just enUS. What if there is no enUS locale available?
- [x] Additionally, how should OnMotdChange be handled? Technically the updated motd can be propagated, but without any further knowledge that could be any locale. I´d argue that the OnMotdChange function should be adjusted to also provide the locale.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
